dao/mongo.go

```diff
@@ -191,3 +207,33 @@ func (m *MongoService) DeletePractitioner(practitionerID string, transactionID s
 
 	return nil, http.StatusNoContent
 }
+
+func (m *MongoService) AppointPractitioner(dao *models.AppointmentResourceDao, transactionID string, practitionerID string) (error, int) {
+
+	collection := m.db.Collection(m.CollectionName)
+
+	// Choose specific practitioner to update
+	filter := bson.M{"transaction_id": transactionID, "data.practitioners.id": practitionerID}
+
+	updateDocument := bson.M{"$set": bson.M{"data.practitioners.$.appointment": dao}}
+
+	update, err := collection.UpdateOne(context.Background(), filter, updateDocument)
+	if err != nil {
+		log.Error(err)
+		return fmt.Errorf("there was a problem handling your request - could not add practitioner appointment to id %s", practitionerID), http.StatusInternalServerError
+	}
+	// Check if Mongo updated the collection
+	if update.MatchedCount == 0 {
+		err = fmt.Errorf("item with transaction id %s or practitioner id %s does not exist", transactionID, practitionerID)
```

Comment:
Would it be better if we queried the database to check an insolvency case with the transaction ID has been retrieved first to provide a detailed error response sent back to the API user (telling them specifically if the case with the transaction ID provided doesn't exist)?

Reply:
I think that would add extra complexity to the code, just for a better error response message. I think I would rather leave as is.

Reply:
Sure, that's fine with me